console/core/src/main/java/org/eclipse/kapua/app/console/core/server/GwtAuthorizationServiceImpl.java

```diff
@@ -72,122 +72,140 @@
-            }
+            // Check Credentials Values
+            ArgumentValidator.notNull(gwtLoginCredentials, "loginCredentials");
+            ArgumentValidator.notEmptyOrNull(gwtLoginCredentials.getUsername(), "loginCredentials.username");
+            ArgumentValidator.notEmptyOrNull(gwtLoginCredentials.getPassword(), "loginCredentials.password");
+
+            // Parse Credentials
+            UsernamePasswordCredentials usernamePasswordCredentials = CREDENTIALS_FACTORY.newUsernamePasswordCredentials(gwtLoginCredentials.getUsername(), gwtLoginCredentials.getPassword());
+            usernamePasswordCredentials.setAuthenticationCode(gwtLoginCredentials.getAuthenticationCode());
+            usernamePasswordCredentials.setTrustKey(gwtLoginCredentials.getTrustKey());
+
+            // Cleanup any previous session
+            cleanupSession();
 
             // Login
-            authenticationService.login(credentials, trustReq);
+            AUTHENTICATION_SERVICE.login(usernamePasswordCredentials, trustReq);
 
-            // Get the session infos
+            // Populate Session
             return establishSession();
         } catch (Throwable t) {
             internalLogout();
-            KapuaExceptionHandler.handle(t);
+
+            throw KapuaExceptionHandler.buildExceptionFromError(t);
         }
-        return null;
     }
 
     @Override
     public GwtSession login(GwtJwtCredential gwtAccessTokenCredentials, GwtJwtIdToken gwtJwtIdToken) throws GwtKapuaException {
-        // VIP
-        // keep this here to make sure we initialize the logger.
-        // Without the following, console logger may not log anything when deployed into tomcat.
-        logger.info(">>> THIS IS INFO <<<");
-        logger.warn(">>> THIS IS WARN <<<");
-        logger.debug(">>> THIS IS DEBUG <<<");
 
         try {
-            // Get the user
-            KapuaLocator locator = KapuaLocator.getInstance();
-            AuthenticationService authenticationService = locator.getService(AuthenticationService.class);
-            CredentialsFactory credentialsFactory = locator.getFactory(CredentialsFactory.class);
-            JwtCredentials credentials = credentialsFactory.newJwtCredentials(gwtAccessTokenCredentials.getAccessToken(), gwtJwtIdToken.getIdToken());
-
-            // Get the session infos
-            if (gwtJwtIdToken == null || gwtJwtIdToken.getIdToken().isEmpty()) {
-                // in this specific case the gwtJwtIdToken cannot be empty
-                throw new KapuaIllegalNullArgumentException("gwtJwtIdToken");
+            // Check Credentials Values
+            ArgumentValidator.notNull(gwtAccessTokenCredentials, "loginCredentials");
+            ArgumentValidator.notNull(gwtAccessTokenCredentials.getAccessToken(), "loginCredentials.accessToken");
+            ArgumentValidator.notNull(gwtJwtIdToken, "jwtIdToken");
+            ArgumentValidator.notNull(gwtJwtIdToken.getIdToken(), "jwtIdToken.idToken");
```

Comment:
Hi @Coduz , I think we should use  `ArgumentValidator.notEmptyOrNull` for the 'jwtIdToken.idToken' in order to mimic the behaviour that was on the old `if` statement.